core/templates/dev/head/player/PlayerServices.js

```diff
@@ -197,31 +196,17 @@ oppia.factory('oppiaPlayerService', [
 
     _stopwatch.resetStopwatch();
 
-    var newStateData = _exploration.states[newStateName];
     $rootScope.$broadcast('playerStateChange');
     successCallback(
       newStateName, refreshInteraction, newFeedbackHtml, newQuestionHtml,
       newParams);
   };
 
-  var _registerMaybeLeaveEvent = function(stateName) {
-    var maybeLeaveExplorationUrl = (
-      '/explorehandler/exploration_maybe_leave_event/' + _explorationId);
-    $http.post(maybeLeaveExplorationUrl, {
-      client_time_spent_in_secs: _stopwatch.getTimeInSecs(),
-      params: learnerParamsService.getAllParams(),
-      session_id: sessionId,
-      state_name: stateName,
-      version: version
-    });
-  };
-
-  // This should only be called when _exploration is non-null.
+  // This should only be called when 'exploration' is non-null.
   var _loadInitialState = function(successCallback) {
-    var initStateName = _exploration.init_state_name;
     var initStateData = stateTransitionService.getInitStateData(
-      _exploration.param_specs, _exploration.param_changes,
-      _exploration.states[initStateName]);
+      exploration.paramSpecs, exploration.paramChanges,
```

Comment:
@BenHenning: I'm thinking of taking the methods in stateTransitionService and bringing them into the Exploration domain object, so that this call can just be exploration.getInitStateData() or similar. This would introduce a dependency in Exploration on expressionInterpolationService, which I think is OK. However, I don't want to introduce a dependency from Exploration to learnerParamsService; I intend the other call to be

     stateTransitionService.getNextStateData(learnerParamsService.getParams())

or similar. What do you think of this plan?